### ISSUE TITLE
Feature: Provide second extensibility build package with esbuild.

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-ui#3215]
Integrate https://github.com/mhsdesign/esbuild-neos-ui-extensibility into the core ~and deprecate the webpack extensibility package~

Related: https://github.com/neos/neos-ui/issues/3113 https://github.com/neos/neos-ui/pull/3200

**Comment by mhsdesign** [comment]
It makes sense to provide both webpack and esbuild as extension point, as they have different features.

See advantages and disadvantages: https://github.com/mhsdesign/esbuild-neos-ui-extensibility#advantages-and-disadvantages-vs-neos-webpack-plugin-system

Also the dynamic import works different in esbuild and we would need https://github.com/neos/neos-ui/issues/3097 for it to work.

currently one installs just one package: `@neos-project/neos-ui-extensibility` and gets webpack.

how can we provide two extension bundlers?

- breaking:
we could make `@neos-project/neos-ui-extensibility` a code only repo and distribute the webpack buildsetup and esbuild plugin separately
eg you need to install:
`@neos-project/neos-ui-extensibility-webpack` or `@neos-project/neos-ui-extensibility-esbuild`
meanwhile the `manifest` is still imported from `@neos-project/neos-ui-extensibility`

- lazy and non breaking:
we could duplicate the code from `@neos-project/neos-ui-extensibility` into a separate package and name it `@neos-project/neos-ui-extensibility-esbuild`
basically just https://github.com/mhsdesign/esbuild-neos-ui-extensibility

"breaking" in terms of -> after a specific version of `@neos-project/neos-ui-extensibility` you dont need to require it anymore - but specify the adapter (esbuild / webpack / ...)

**Comment by mhsdesign** [comment]
closed via: https://github.com/neos/neos-ui/pull/3382 now it only needs to be documented via https://github.com/neos/neos-ui/issues/3445
